### PR TITLE
[EOS-19670] Parsing of extended metadata key, since object name can contain delimiter |

### PR DESCRIPTION
**Pull request by DPG17**
- Changed the parsing logic
- Also, removed code that relates to "~" as start object prefix used in earlier design of
  S3 fault tolerance feature.

Signed-off-by: Dattaprasad <dattaprasad.govekar@seagate.com>

### CODE REVIEW

File: server/s3_put_multiobject_action.cc [hunk 1]
```diff
@@ -144,7 +144,7 @@ void S3PutMultiObjectAction::check_part_details() {
   } else if (request->get_header_size() > MAX_HEADER_SIZE ||
              request->get_user_metadata_size() > MAX_USER_METADATA_SIZE) {
     s3_put_action_state = S3PutPartActionState::validationFailed;
-    set_s3_error("BadRequest");
+    set_s3_error("MetadataTooLarge");
```
Comment (DPG17) [inline server/s3_put_multiobject_action.cc:147]: This change is unrelated to this ticket. It was done to fix UT

**Review by knrajnambiar76 — APPROVED** [review]

**Review by nileshgovande — APPROVED** [review]
